src/Components/Endpoints/src/Builder/RazorComponentsEndpointConventionBuilder.cs

```diff
@@ -103,4 +72,13 @@ public void Finally(Action<EndpointBuilder> finallyConvention)
             _finallyConventions.Add(finallyConvention);
         }
     }
+
+    /// <summary>
+    /// Adds the given <paramref name="renderMode"/> to the list of configured render modes if not present.
+    /// </summary>
+    /// <param name="renderMode">The <see cref="IComponentRenderMode"/> to add.</param>
+    public void AddRenderMode(IComponentRenderMode renderMode)
```

Comment:
```suggestion
    public RazorComponentsEndpointConventionBuilder AddRenderMode(IComponentRenderMode renderMode)
```

Wasn't there supposed to be `AddWebAssemblyRenderMode()` directly on the `RazorComponentsEndpointConventionBuilder` according to the approved API proposal at https://github.com/dotnet/aspnetcore/issues/49754?

I don't think we approved any `IComponentRenderMode` APIs other than `RenderModeEndpointProvider` which we tried to hide in an Infrastructure namespace. I don't think `RenderMode` or the `IComponentRenderMode` interface itself was ever proposed as API either.

I see that this is being called by an `AddWebAssemblyRenderMode` extension method in `WebAssemblyRazorComponentsEndpointConventionBuilderExtensions`. I'm assuming this was done for layering reasons so we could flow options via the call to `AddWebAssemblyRenderMode(WebAssemblyComponentsEndpointOptions)`.

It's really odd to use different patterns for adding server and webassembly render modes. If we're going to do an extension method for one, we should do it for both. It's also a bit unusual to take an `WebAssemblyComponentsEndpointOptions` directly instead of an `Action<WebAssemblyComponentsEndpointOptions>`, but maybe it's okay since we're not really doing the options pattern for endpoint config. Regardless, it's something that should be discussed in API review.